rust/xaynet-server/src/state_machine/phases/sum2.rs

```diff
@@ -120,47 +107,38 @@ impl PhaseState<Sum2> {
         Self {
             inner: Sum2 {
                 model_agg,
-                mask_dict: MaskDict::new(),
+                sum2_count: 0,
             },
             shared,
         }
     }
 
-    /// Handles a sum2 request.
-    /// If the handling of the sum2 message fails, an error is returned to the request sender.
-    fn handle_sum2(&mut self, req: Sum2Request) -> Result<(), StateMachineError> {
+    /// Handles a sum2 request by adding a mask to the mask dictionary.
+    ///
+    /// # Errors
+    /// Fails if the sum participant didn't register in the sum phase or it is a repetition.
+    async fn handle_sum2(&mut self, req: Sum2Request) -> Result<(), StateMachineError> {
         let Sum2Request {
             participant_pk,
             model_mask,
         } = req;
-        self.add_mask(&participant_pk, model_mask)
-    }
 
-    /// Adds a mask to the mask dictionary.
-    ///
-    /// # Errors
-    /// Fails if the sum participant didn't register in the sum phase or it is a repetition.
-    fn add_mask(
-        &mut self,
-        _pk: &SumParticipantPublicKey,
-        mask: MaskObject,
-    ) -> Result<(), StateMachineError> {
-        if let Some(count) = self.inner.mask_dict.get_mut(&mask) {
-            *count += 1;
-        } else {
-            self.inner.mask_dict.insert(mask, 1);
-        }
+        self.shared
+            .io
+            .redis
+            .connection()
+            .await
```

Comment:
I know this is not new from this PR but it didn't strike me before: why do we need to `await` for the connection? Don't we have a pool of pre-established connections?

Reply:
The `await` comes form the `Semaphore` that limits the number of concurrent requests. However, the multiplex connection is only a [single connection](https://github.com/mitsuhiko/redis-rs/issues/218#issuecomment-509216100).
For connection pool we would need something like [`deadpool-redis`](https://crates.io/crates/deadpool-redis) or [`bb8-redis`](https://crates.io/crates/bb8-redis/0.3.1) ([ref of the comment](https://github.com/mitsuhiko/redis-rs/pull/388#issuecomment-702239648))

Reply:
I see. All good then. I was just afraid we were re-establishing a new connection every time.